<launch>
    
    <group>
        <include file="$(find-pkg-share pingpongbot_description)/view_robot.launch.xml" >
            <arg name="use_jsp" value="none"/>
        </include>
    </group>

    <node pkg="pingpongbot_control" exec="jointStateUpdate">
        <remap from="wheel_speeds" to="wheel_speeds"/>
        <remap from="wheel_angles" to="wheel_angles"/>
        <remap from="joint_states" to="joint_states"/>
    </node>

    <node pkg="pingpongbot_control" exec="odomUpdate">
        <remap from="wheel_speeds" to="wheel_speeds"/>
        <remap from="wheel_angles" to="wheel_angles"/>
        <remap from="joint_states" to="joint_states"/>
        <remap from="cmd_vel" to="cmd_vel"/>
        <param from="$(find-pkg-share pingpongbot_control)/robot_dimensions.yaml"/>
    </node>

    <node pkg="pingpongbot_control" exec="controller">
        <remap from="cmd_vel" to="cmd_vel"/>
        <remap from="goal_pose" to="goal_pose"/>
        <param from="$(find-pkg-share pingpongbot_control)/controller.yaml"/>
    </node>

</launch>